### ISSUE TITLE
dependsOn is not ignoring hidden fields

### ISSUE THREAD

**Issue by williamcruzme** [laravel/nova-issues#4170]
- Laravel Version: 9.9.0
- Nova Version: 4.4.1
- PHP Version: 8.0
- Database Driver & Version: MySQL
- Operating System and Version: MacOS 11.6
- Browser type and version: Chrome 101.0.4951.54

### Issue 1:
Having a field like:
```php
Select::make('Merchant', 'merchant_id')
    ->options(Merchant::pluck('official_name', 'id'))
    ->displayUsingLabels()
    ->onlyOnForms()
    ->dependsOn(
        ['type'],
        function ($field, $request, $formData) {
            $field->hide();
        }
    ),
```

It's still including the `merchant_id` in the Body with a value.

<img width="289" alt="image" src="https://user-images.githubusercontent.com/22250407/167021112-b8bac8ff-f075-47d9-a376-37cf962414db.png">

### Issue 2:
Having hidden field like:
```php
Number::make('Amount')
    ->rules('required')
    ->dependsOn(
        ['type'],
        function ($field, $request, $formData) {
            $field->hide();
        }
    )
```

It's still reading the rules:
<img width="708" alt="image" src="https://user-images.githubusercontent.com/22250407/167021519-ada6d212-7202-46f3-b343-4d7ee78c0ee9.png">

-----

Those issues are new. That wasn't happening.

**Comment by crynobone** [comment]
There's a known issue with `hide` and `show` options and currently working on a fix. However you still can't use `rules('required')` with `hide()`.

**Comment by benedict-w** [comment]
Did this get fixed?

I'm trying to show / hide a field on a checkbox, but am getting "Uncaught (in promise)" JS errors:

```
    Boolean::make('Web Safe', 'web_safe')
        ->default(true),

    File::make('TTF', 'formats->ttf')
        ->rules(['mimes:ttf'])
        ->disk('public')
        ->hide()
        ->dependsOn(
            ['web_safe'],
            function (File $field, NovaRequest $request, FormData $formData) {
                if(!$formData->web_safe) {
                    $field->show();
                }
            }
        ), 
```

**Comment by davidhemphill** [comment]
There was a fix merged and released in 4.7.0.
